### ISSUE TITLE
RuntimeWarning: coroutine 'Client._close' was never awaited

### ISSUE THREAD

**Issue by Keou0007** [dask/distributed#3785]
I'm using a wrapper function to add semaphore functionality to a function I want to run in parallel and it's causing RuntimeWarnings to appear when I close the Client. Aside from the warnings, it seems to be working just how I intended.

Using dask version 2.15.0 and distributed version 2.15.2

Minimal reproducible example:
```
import time
import logging
import sys
import tracemalloc
from dask.distributed import Client, Semaphore


logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
tracemalloc.start()


def test(index):
    time.sleep(1)
    print(f"{index} done")


def wrapper(func, sem):
    def inner(*args, **kwargs):
        with sem:
            return func(*args, **kwargs)
    return inner


def main():
    client = Client(
        n_workers=4,
        threads_per_worker=1,
        dashboard_address=None,
    )
    sem = Semaphore(max_leases=2, client=client)
    futures = client.map(wrapper(test, sem), range(10))
    client.gather(futures)
    sem.close()
    client.close()


if __name__ == '__main__':
    main()
```

output:

```
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:asyncio:Using selector: KqueueSelector
2 done
3 done
1 done
0 done
6 done
7 done
5 done
4 done
9 done
8 done
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'wait_for' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1357
    future = asyncio.wait_for(future, timeout)
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'wait_for' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1357
    future = asyncio.wait_for(future, timeout)
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'Client._close' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1355
    future = self._close()
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'wait_for' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1357
    future = asyncio.wait_for(future, timeout)
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'Client._close' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1355
    future = self._close()
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'Client._close' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1355
    future = self._close()
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'wait_for' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1357
    future = asyncio.wait_for(future, timeout)
/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'Client._close' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/anaconda3/lib/python3.7/site-packages/distributed/client.py", lineno 1355
    future = self._close()

Process finished with exit code 0

```

**Comment by lr4d** [comment]
I guess this should be fixed by https://github.com/dask/distributed/pull/3713

**Comment by Keou0007** [comment]
You're right, that appears to be the fix.

**Comment by Keou0007** [comment]
I've updated to 2.18 and I'm still seeing this. I didn't pay close enough attention before, but the other PR just fixes the issue in semaphore, what I'm seeing here is in client.

<details>
<summary> Traceback </summary >

```
/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'wait_for' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py", lineno 1357
    future = asyncio.wait_for(future, timeout)
/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'wait_for' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py", lineno 1357
    future = asyncio.wait_for(future, timeout)
/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'Client._close' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py", lineno 1355
    future = self._close()
/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py:4721: RuntimeWarning: coroutine 'Client._close' was never awaited
  c.close(timeout=2)
Object allocated at (most recent call last):
  File "/Users/shannon/opt/miniconda3/envs/pywakes/lib/python3.8/site-packages/distributed/client.py", lineno 1355
    future = self._close()
```
</details>

**Comment by Keou0007** [comment]
@fjetter this issue needs to be re-opened, but I can't because you closed it.

**Comment by fjetter** [comment]
Indeed, this is still not fixed. Issue comes from the client which is attached to the worker. Will open another PR for this
